fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -165,13 +165,17 @@ def self.keychain_path(name)
       # has changed for some users: https://github.com/fastlane/fastlane/issues/5649
       #
 
+      # Remove the ".keychain" at the end of the name
+      name.sub!(/\.keychain$/, "")
+
       keychain_paths = [
         File.join(Dir.home, 'Library', 'Keychains', name),
-        File.join(Dir.home, 'Library', 'Keychains', "#{name}-db"),
-        name,
-        "#{name}-db"
+        name
       ].map { |path| File.expand_path(path) }
 
+      # Transforms ["thing"] to ["thing", "thing-db", "thing.keychain", "thing-db.keychain"]
+      keychain_paths = keychain_paths.product(["", "-db"], ["", ".keychain"]).map(&:join)
```

Comment:
This is definitely some sweet sweet functional awesomeness, but given that there are only 8 options, I'm wondering if it would be more explicit for contributors to come in and understand the code if we just spelled out each option in the array above?